dmoj/executors/RUST.py

```diff
@@ -56,15 +57,40 @@ def create_files(self, problem_id, source_code, *args, **kwargs):
         with open(self._file('Cargo.toml'), 'wb') as f:
             f.write(CARGO_TOML)
 
+    shared_target = None
+
+    @classmethod
+    def get_shared_target(cls):
+        if cls.shared_target is None:
+            cargo_dir = os.path.expanduser('~/.cargo')
+            collisions = 0
+            while True:
+                maybe_target = cargo_dir + f'/dmoj-shared-target-{collisions}'
+                try:
+                    os.mkdir(maybe_target, mode=0o775)
+                except FileExistsError:
+                    pass
+                try:
+                    # This is 'leaked' on purpose.
+                    dirfd = os.open(maybe_target, os.O_RDONLY | os.O_CLOEXEC)
+                    fcntl.flock(dirfd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                except BlockingIOError:
+                    collisions += 1
```

Comment:
We shouldn't leak dirfd in this branch.

Reply:
Agreed, and honestly, we shouldn't have it in the `try` at all.